lib/src/config/config.rs

```diff
@@ -847,7 +859,20 @@ impl ClientConfigBuilder {
         // Configure validator
         #[cfg(feature="validator")] {
             if config_file.validator.is_some() {
-                self.validator = Some(Some(ValidatorConfig {}));
+                let wallet_private_key = config_file.validator.as_ref()
+                    .map(|settings| settings.clone().wallet_private_key)
+                    .expect("Failed to load validator settings");
+
+                let wallet_key = wallet_private_key.map(|config_private_key| {
+                    Some(keys::KeyPair::from(
+                        keys::PrivateKey::from_str(&config_private_key)
+                            .map_err(|e| Error::config_error(format!("Invalid wallet private key: {:?}: {}", &config_private_key, e))).ok()?
+                    ))
+                }).unwrap_or_else(|| None);
```

Comment:
Just FYI, you don't need this. `wallet_private_key` is already an `Option` and `unwrap_or_else` is called when this is `None`, replacing it with a `None`. So, basically a no-op. :)